### PR TITLE
Retire dolphin runtime

### LINKED ISSUE #958: Retire Dolphin Runtime
Our testnet is now using calamari RT and Dolphin does not represent the state of the chain due to e.g. missing staking

Can we retire this RT codebase altogether or do we need to keep it around?
Benefits of eliminating the code should be obvious, drawbacks not so much.

We'd need to merge dolphin and calamari integration tests

### PR DESCRIPTION
**Pull request by ghzlatarev**
## Description

closes: #958 

* Retire all Dolphin runtime related code, as it is not longer used for anything.
* check-tests was excluding pallet-manta-pay tests, when we only wanted to exclude the benchmark unit test (as our code relies on externally passed chain-spec file) . The fix is to not exclude in the CI but just hard-code disable the benchmark unit tests in the pallet code.
* gov v2 imports are required because of upstream code, so moved them outside of integrations test crate, as that crate is excluded in our unit tests job and it will fail compilation

---

Before we can approve this PR for merge, please make sure that **all** the following items have been checked off:
- [x] Connected to an issue with discussion and accepted design using zenhub "Connect issue" button below
- [x] Added **one** label out of the `L-` group to this PR
- [x] Added **one or more** labels from the `A-` and `C-` groups to this PR
- [x] Explicitly labelled `A-calamari` and/or `A-manta` if your changes are meant for/impact either of these (CI depends on it)
- [x] Re-reviewed `Files changed` in the Github PR explorer.

Situational Notes:
- If adding functionality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.

### CODE REVIEW

**Comment by ghzlatarev** [comment]
cargo clippy seems to be broken with any rust version

**Comment by Dengjianping** [comment]
And we might delete dolphin docs from our manta docs

**Comment by Dengjianping** [comment]
And some dolphin genesis files should be deleted

**Review by ferrell-code — COMMENTED** [review]
looks good ci still seems to be failing

**Comment by ghzlatarev** [comment]
> looks good ci still seems to be failing

Yeah i see why.
Something from upstream pulls the gov v2 pallets, which require us to also include them.
But if we include them in the integrations-test crate (and exclude it in the unit tests) then it won't compile.

This actually showed that we had broken our unit tests in another way. We wanted to exclude the manta-pay benchmarking unit test, but we had excluded all the unit tests. Fixed this now.

File: runtime/calamari/Cargo.toml [hunk 1]
```diff
@@ -59,6 +59,10 @@ pallet-treasury = { git = "https://github.com/paritytech/substrate.git", default
 pallet-utility = { git = 'https://github.com/paritytech/substrate.git', default-features = false, branch = "polkadot-v0.9.37" }
 substrate-fixed = { git = "https://github.com/Manta-Network/substrate-fixed.git", default-features = false }
 
+pallet-conviction-voting = { git = "https://github.com/paritytech/substrate.git", default-features = false, branch = "polkadot-v0.9.37" }
```
Comment (Dengjianping) [inline runtime/calamari/Cargo.toml:62]: I know why you import these three pallet here, it's kind of annoying.

**Review by Dengjianping — APPROVED** [review]

**Comment by Dengjianping** [comment]
I thought we would have faster ci check once dolphin runtime.
By the way, not sure it will affect change log generation, the dolphin tag `[DO]`. I suppose it won't.

File: runtime/calamari/Cargo.toml (hunk 1 above)
Comment (ghzlatarev) [reply runtime/calamari/Cargo.toml:62]: yeah i just moved them from the integrations tests crate because it is excluded in some of our tests

Comment (ghzlatarev) [reply runtime/calamari/Cargo.toml:62]: added to the description

**Comment by ghzlatarev** [comment]
> I thought we would have faster ci check once dolphin runtime. By the way, not sure it will affect change log generation, the dolphin tag `[DO]`. I suppose it won't.

no, we stopped Dolphin jobs some time ago already. This is just to reduce code base -8000 lines feels good

also re-includes mantapay in the unit tests job as they were not running for a few days

**Review by Apokalip — APPROVED** [review]

**Pull request merged**